app/javascript/dashboard/routes/dashboard/settings/teams/AgentSelector.vue

```diff
@@ -0,0 +1,179 @@
+<template>
+  <div>
+    <div class="add-agents__header"></div>
+    <table class="woot-table">
+      <thead>
+        <tr>
+          <td>
+            <div class="checkbox-wrap">
+              <input
+                name="select-all-agents"
+                type="checkbox"
+                :checked="allAgentsSelected ? 'checked' : ''"
+                :title="$t('TEAMS_SETTINGS.AGENTS.SELECT_ALL')"
+                @click.self="selectAllAgents"
+              />
+            </div>
+          </td>
+          <td>{{ $t('TEAMS_SETTINGS.AGENTS.AGENT') }}</td>
+          <td>{{ $t('TEAMS_SETTINGS.AGENTS.EMAIL') }}</td>
+        </tr>
+      </thead>
+      <tbody>
+        <tr
+          v-for="agent in agentList"
+          :key="agent.id"
+          :class="agentRowClass(agent.id)"
+        >
+          <td class="checkbox-cell">
+            <div class="checkbox-wrap">
+              <input
+                type="checkbox"
+                :checked="isAgentSelected(agent.id)"
+                @click.self="() => handleSelectAgent(agent.id)"
```

Comment:
Why is .self and why is a method a defined here? Can't we set a value to the input and get that value from event.target?

Reply:
Will have to do manual type casting with event.target since id is number